topostats/plottingfuncs.py

```diff
@@ -604,6 +604,8 @@ def save_figure(self):
                         self.pixel_to_nm_scaling,
                         (2, -2),
                     )
+            if not self.output_dir.exists():
```

Comment:
Given you have the `.mkdir(parents=True, exist_ok=True)` parameters its redundant to check whether the directory already exists before making the directory.
```suggestion
```